sympy/core/evalf.py

```diff
@@ -1391,6 +1391,44 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             verbose=<bool>
                 Print debug information (default=False)
 
+        Note:
+        It is suggested to use expr.evalf(subs=...) over expr.subs(...).evalf()
+        or expr.evalf().subs(...) since expr.evalf(subs=...) avoids the loss of
+        significance caused due to naive substitution as the expression is run
+        through the evalf algorithm which takes into account, the various issues
+        that can cause loss of significance.
+
+        Examples
+        ========
+
+        >>> import sympy
+        >>> omega, t = sympy.symbols("omega, t")
+        >>> s = sympy.pi * sympy.cos(omega*t)
+
+        >>> print(s.evalf().subs({t: 0}))
+        3.14159265358979
+
+        >>> print(s.subs({t: 0}).evalf())
+        3.14159265358979
+
+        >>> print(s.evalf(subs={t: 0})) #let's call this to_do_1
+        3.14159265358979*cos(omega*t)
+
+        in to_do_1 evalf(subs...) evalf leaves the expression as it is, in order to
```

Comment:
It would be better not to document the bug (#6974); even if `omega` is finite this substitution doesn't happen. It doesn't happen because the subs mechanism is broken for Mul because of how evalf is defined.

Reply:
@smichr  so should I remove the bug example ?